### PR TITLE
Prevent segfault in vtk2mesh

### LINKED ISSUE #4487: [io]  loadPolygonFilePLY segfaults when file does not exist
This code
```
  PolygonMesh wat;
  io::loadPolygonFilePLY("n/a", wat);

```

produces a segfault in PCL 1.11.1, so there is no way to recover from the case that the file does not exist.

Some stuff is printed to console:
```
Warning: In /tmp/vtk@8.2-20200909-79228-1qm7d8t/VTK-8.2.0/IO/PLY/vtkPLYReader.cxx, line 132
vtkPLYReader (0x7ff83dc1a3f0): Could not open PLY file

ERROR: In /tmp/vtk@8.2-20200909-79228-1qm7d8t/VTK-8.2.0/Common/ExecutionModel/vtkExecutive.cxx, line 782
vtkCompositeDataPipeline (0x7ff83dc1aa70): Algorithm vtkPLYReader(0x7ff83dc1a3f0) returned failure for request: vtkInformation (0x7ff83dc1b3b0)
  Debug: Off
  Modified Time: 118
  Reference Count: 1
  Registered Events: (none)
  Request: REQUEST_DATA
  FROM_OUTPUT_PORT: 0
  ALGORITHM_AFTER_FORWARD: 1
  FORWARD_DIRECTION: 0

Segmentation fault: 11
```

**System:**

- osx 10.15
- PCL 1.11.1
- vtk 8.2

### PR DESCRIPTION
**Pull request by mvieth**
Can e.g. happen if non-existent file is read
Fixes #4487
@themightyoarfish

### CODE REVIEW

File: io/src/vtk_lib_io.cpp [hunk 1]
```diff
@@ -240,6 +240,11 @@ pcl::io::vtk2mesh (const vtkSmartPointer<vtkPolyData>& poly_data, pcl::PolygonMe
   mesh.cloud.width = mesh.cloud.height = 0;
   mesh.cloud.is_dense = true;
 
+  if (poly_data->GetPoints () == nullptr)
```
Comment (kunaltyagi) [inline io/src/vtk_lib_io.cpp:243]: Should we even resize the mesh? Shouldn't we perform the check as the first thing in the function? The use of 0 for return value can be used at call-sites to check if the call failed.

Comment (mvieth) [reply io/src/vtk_lib_io.cpp:243]: I think it is safer to initialize `mesh` first. If there was something else in `mesh` which is not overwritten, then this might lead to confusion.

Comment (kunaltyagi) [reply io/src/vtk_lib_io.cpp:243]: `mesh` is a reference, so it might be safe to do nothing, but better safe than sorry

**Review by kunaltyagi — APPROVED** [review]

File: io/src/vtk_lib_io.cpp [hunk 2]
```diff
@@ -240,6 +240,11 @@ pcl::io::vtk2mesh (const vtkSmartPointer<vtkPolyData>& poly_data, pcl::PolygonMe
...
+  {
+    PCL_ERROR ("[pcl::io::vtk2mesh] Given vtkPolyData does not contain any points.\n");
```
Comment (larshg) [inline io/src/vtk_lib_io.cpp:245]: Maybe rephrase to include nullptr - since, "contain any points", could just mean it was initialized correctly, but with 0 points.

**Review by larshg — APPROVED** [review]
